File: 2015/01.R
source(file.path(getwd(), '2015/utils.R'))
data <- getInputData(1)

solvePartOne <- function() {
  ups <- stringr::str_count(data, '\\(')
  downs <- stringr::str_count(data, '\\)')
  return(ups - downs)
}

solvePartTwo <- function() {
  inputData <- strsplit(data, '')[[1]]
  indexVector <- seq_along(inputData)
  floorVector <- cumsum(ifelse(inputData == '(', 1, -1))
  return(min(indexVector[floorVector == -1]))
}

cat('Part One:', solvePartOne(), '\n')
cat('Part Two:', solvePartTwo(), '\n')

File: 2018/01.js
import {getInputData} from './utils.js';

const data = (await getInputData(1)).split('\n').filter(x => x !== '').map(Number);

function calculateFrequency(stopOnRepeat = false) {
	let frequency = 0;
	const seen = new Set([0]);
	let index = 0;
	while (true) {
		frequency += data[index];
		if (stopOnRepeat && seen.has(frequency)) return frequency;
		seen.add(frequency);
		index = (index + 1) % data.length;
		if (!stopOnRepeat && index === 0) break;
	}
	return frequency;
}

console.log(`Part One: ${calculateFrequency()}`);
console.log(`Part Two: ${calculateFrequency(true)}`);